# ML-02 — Research Question and Provisional Lane

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

I am choosing the CTR / Engagement Opportunity Scoring lane because it turns a clear product question into a practical ranking problem: which pages are visible enough to matter, but are not earning clicks as strongly as they should? That makes the work useful for content editors and SEO specialists who need a short list of pages worth reviewing first, and it fits the next seven weeks because the starter data already contains the core signals we need: impressions, clicks, CTR, average position, and content age.


In [2]:
import pandas as pd

# Load the starter dataset.
df = pd.read_csv("D:/Flyrank/Flyrank-assignments/data/raw/content_refresh_anonymized.csv")

# Keep the answer concise and evidence-based.
response = (
    "For content editors deciding which pages to review first, I will build a ranked "
    "opportunity score from observed search and engagement signals such as impressions, "
    "clicks, CTR, average position, and content freshness. A wrong recommendation costs "
    "editor time and can cause a worthwhile page to be missed, so the output should be "
    "treated as decision-support rather than a guarantee."
)

# print("Selected Lane: CTR / Engagement Opportunity Scoring")
response

'For content editors deciding which pages to review first, I will build a ranked opportunity score from observed search and engagement signals such as impressions, clicks, CTR, average position, and content freshness. A wrong recommendation costs editor time and can cause a worthwhile page to be missed, so the output should be treated as decision-support rather than a guarantee.'

## 2. The question: decision, action, cost of a wrong call

The decision is: which pages should an editor prioritize for a CTR-focused review first? The action is a review or refresh recommendation for a specific page. A wrong call mainly wastes editor time and can delay a useful improvement on a page that actually has opportunity. The work is useful because it can help turn a large backlog into a shorter, ranked review queue, but it should stay decision-support rather than a claim that we can predict Google itself.


In [3]:
import pandas as pd

# Load the starter dataset.
df = pd.read_csv("d:/Flyrank/Flyrank-assignments/data/raw/content_refresh_anonymized.csv")

# Show a simple, concrete framing of the decision.
summary = {
    "rows": len(df),
    "pages_with_impressions_over_1000": int((df["impressions_90d"] >= 1000).sum()),
    "pages_with_clicks": int((df["clicks_90d"] > 0).sum()),
    "pages_with_ctr_above_0_3": int((df["ctr"] > 0.3).sum()),
}
summary

{'rows': 30000,
 'pages_with_impressions_over_1000': 13512,
 'pages_with_clicks': 16796,
 'pages_with_ctr_above_0_3': 7035}

## 3. Quick look at the data (2-3 real numbers)

The starter dataset already shows enough signal to justify the CTR lane. In this data, 13,512 pages had at least 1,000 impressions over 90 days, 7,035 pages had CTR above 0.3, and the median CTR was 0.07. Those figures suggest there is a meaningful spread in click performance, which makes a ranking-style opportunity score worth exploring.


In [4]:
import pandas as pd

# Load the starter dataset.
df = pd.read_csv("d:/Flyrank/Flyrank-assignments/data/raw/content_refresh_anonymized.csv")

# Real numbers that support the CTR lane.
metrics = {
    "median_ctr": round(float(df["ctr"].median()), 3),
    "pages_with_ctr_above_0_3": int((df["ctr"] > 0.3).sum()),
    "pages_with_impressions_over_1000": int((df["impressions_90d"] >= 1000).sum()),
}
metrics

{'median_ctr': 0.07,
 'pages_with_ctr_above_0_3': 7035,
 'pages_with_impressions_over_1000': 13512}

## 4. Careful words: what I can and can't claim

I can claim that this dataset shows observed patterns in impressions, clicks, CTR, position, and freshness that are useful for ranking pages by apparent opportunity. I can also say that a model or scoring rule can be used as decision-support for a review queue. I cannot claim that the work proves causation, that it predicts Google behavior, or that it will guarantee a click increase for a refreshed page.


In [5]:
import pandas as pd

# Load the starter dataset.
df = pd.read_csv("d:/Flyrank/Flyrank-assignments/data/raw/content_refresh_anonymized.csv")

# Keep the claims carefully bounded.
claims = {
    "can_claim": "Observed patterns in this dataset support a decision-support ranking of CTR opportunity.",
    "cannot_claim": "This work does not prove causation or predict Google's algorithm.",
}
claims

{'can_claim': 'Observed patterns in this dataset support a decision-support ranking of CTR opportunity.',
 'cannot_claim': "This work does not prove causation or predict Google's algorithm."}

## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.